src/clients.rs

```diff
@@ -202,24 +203,19 @@ struct InnerState {
 
 impl State {
     /// Creates a new state instance. Should only be called once.
-    ///
-    /// # Panics
-    /// If called before [`set_cache_dir`].
-    pub fn new(client: RuskStateClient) -> Result<Self, Error> {
-        let cache = Cache::new()?;
+    pub fn new(
+        client: RuskStateClient,
+        data_dir: &Path,
+    ) -> Result<Self, Error> {
+        let cache = Cache::new(data_dir)?;
         let inner = Mutex::new(InnerState { client, cache });
```

Comment:
As a side note, if this is called in `async` code, shouldn't we use `tokio::Mutex` so we yield instead of blocking the thread?

Reply:
It's generally advised from the tokio community to use the std `Mutex` because there's no reason for the Mutex to not block, it usually should block the thread its in so there's no race conditions. 

`tokio::Mutex` should only be used if we want to lock the mutex **across awaits**

@goshawk-3

Reply:
That's a wrong statement.  `sync::Mutex` should not be used in async code ever as it will not allow Tokio::Scheduler to yield. As a result this blocks a thread from the pool ( in an edge case you can use spawn_blocking to handle this)

There will be no race condition if you use tokio::Mutex if you don't mix async and sync calls.

Reply:
FYI  https://docs.rs/tokio/1.28.1/tokio/index.html#cpu-bound-tasks-and-blocking-code
@Daksh14

Reply:
@goshawk-3 
>  sync::Mutex should not be used in async code ever as it will not allow Tokio::Scheduler to yield

That is an incorrect statement 

https://docs.rs/tokio/latest/tokio/sync/struct.Mutex.html#which-kind-of-mutex-should-you-use 

> Contrary to popular belief, it is ok and often preferred to use the ordinary [Mutex](https://doc.rust-lang.org/nightly/std/sync/mutex/struct.Mutex.html) from the standard library in asynchronous code.

> The feature that the async mutex offers over the blocking mutex is the ability to keep it locked across an **.await point. **...